File: TravelWise-backend/src/main/java/com/example/travelwise/entity/TravelerDetails.java
package com.example.travelwise.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class TravelerDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String travelerFirstName;
    private String travelerSurname;
    private String travelerEmail;
    private String travelerPhoneNumber;
    @ManyToOne
    @JoinColumn(name = "reservation_id")
    private Reservation reservation;
}

File: TravelWise-frontend/src/app/models/travelerDetails.ts
export class TravelerDetails {
  id?: null;
  travelerFirstName: string;
  travelerSurname: string;
  travelerEmail: string;
  travelerPhoneNumber: string;
  reservationId?: number;

  constructor(travelerFirstName: string, travelerSurname: string, travelerEmail: string, travelerPhoneNumber: string) {
    this.travelerFirstName = travelerFirstName;
    this.travelerSurname = travelerSurname;
    this.travelerEmail = travelerEmail;
    this.travelerPhoneNumber = travelerPhoneNumber;
  }
}
